src/sagemaker/jumpstart/notebook_utils.py

```diff
@@ -283,166 +310,146 @@ def _generate_jumpstart_model_versions(  # pylint: disable=redefined-builtin
-                model_filters,
+        if copied_filter.resolved_value == BooleanValues.UNEVALUATED:
+            raise RuntimeError(
+                "Filter expression in unevaluated state after using "
+                "values from model manifest. Model ID and version that "
+                f"is failing: {(model_manifest.model_id, model_manifest.version)}."
             )
+        copied_filter_2 = copy.deepcopy(filter)
 
-            _put_resolved_booleans_into_filter(copied_filter, model_filters_to_resolved_values)
-
-            copied_filter.eval()
-
-            if copied_filter.resolved_value in [BooleanValues.TRUE, BooleanValues.FALSE]:
-                if copied_filter.resolved_value == BooleanValues.TRUE:
-                    yield (model_manifest.model_id, model_manifest.version)
-                continue
-
-            if copied_filter.resolved_value == BooleanValues.UNEVALUATED:
-                raise RuntimeError(
-                    "Filter expression in unevaluated state after using "
-                    "values from model manifest. Model ID and version that "
-                    f"is failing: {(model_manifest.model_id, model_manifest.version)}."
+        model_specs = JumpStartModelSpecs(
+            json.loads(
+                sagemaker_session.read_s3_file(
+                    get_jumpstart_content_bucket(region), model_manifest.spec_key
                 )
-            copied_filter_2 = copy.deepcopy(filter)
-
-            model_specs = accessors.JumpStartModelsAccessor.get_model_specs(
-                region=region,
-                model_id=model_manifest.model_id,
-                version=model_manifest.version,
             )
+        )
 
-            model_specs_keys = set(model_specs.__slots__)
+        for val in possible_spec_keys:
+            if hasattr(model_specs, val):
+                manifest_specs_cached_values[val] = getattr(model_specs, val)
 
-            unrecognized_keys -= model_specs_keys
-            unrecognized_keys_for_single_spec = possible_spec_keys - model_specs_keys
-            unrecognized_keys.update(unrecognized_keys_for_single_spec)
+        _populate_model_filters_to_resolved_values(
+            manifest_specs_cached_values,
+            model_filters_to_resolved_values,
+            model_filters,
+        )
+        _put_resolved_booleans_into_filter(copied_filter_2, model_filters_to_resolved_values)
 
-            for val in possible_spec_keys:
-                if hasattr(model_specs, val):
-                    manifest_specs_cached_values[val] = getattr(model_specs, val)
+        copied_filter_2.eval()
 
-            _populate_model_filters_to_resolved_values(
-                manifest_specs_cached_values,
-                model_filters_to_resolved_values,
-                model_filters,
-            )
-            _put_resolved_booleans_into_filter(copied_filter_2, model_filters_to_resolved_values)
+        if copied_filter_2.resolved_value != BooleanValues.UNEVALUATED:
```

Comment:
you don't need this `if` based on your conditional on line 418.

Also should like 418 read: `or (copied_filter_2.resolved_value == BooleanValues.UNKNOWN and list_incomplete_models)` ?

Reply:
Don't we need it in case it ends up in an unevaluated state (runtime error)?